doc/schema/schema.md

```diff
@@ -95,7 +95,7 @@ Parameters of the sampling algorithm and sampling backend to be used for analysi
 
 ### Out of sample `posterior_predictive` samples
 #### `predictions`
-Out of sample posterior predictive samples p(y'|y). Sample should match `posterior` samples. Its variables should have a counterpart in `posterior_predictive`. However, variables in `predictions` and their counterpart in `posterior_predictive` may not share coordinates.
+Out of sample posterior predictive samples p(y'|y). Samples should match `posterior samples. Its variables should have a counterpart in `posterior_predictive`. However, variables in `predictions` and their counterpart in `posterior_predictive` can have different coordinates.
```

Comment:
Missing close backtick around "posterior".

To further clarify, suggest we change the final phrase to "can have different coordinate values.  E.g., the posterior might have values `[1, 2, 3]` along coordinate `x`, while the posterior predictive might have `[4, 5, 6]`."

I like the simple name "predictions" as an alternative to "out of sample posterior predictive samples."  That's an improvement!

Reply:
Thanks. I would change the example wording in order to keep the notation: "e.g., the posterior predictive might have coordinates `[0, 1, 2]` along dimension `x`, while the predictions may have `[3, 4, 5]`". 

As I understand xarray notation, `x` is not a coordinate but a dimension.

Reply:
In xarray output, the first thing to be printed are the dimensions, then the coordinates (which in arviz are assigned to a dimension but in general xarray objects, it is possible to have coordinates without dimension assigned) and eventuallt data variables which have some dimensions assigned

Reply:
> Thanks. I would change the example wording in order to keep the notation: "e.g., the posterior predictive might have coordinates `[0, 1, 2]` along dimension `x`, while the predictions may have `[3, 4, 5]`".

If you are going to introduce the posterior predictive in this way, I think we should address all three:

"e.g., the posterior distribution and the posterior predictive might have coordinates `[0, 1, 2]` along dimension `x`, while the predictions might have `[3, 4, 5]`".

This captures the distinction between posterior predictive and predictions.  `[2, 3, 4, 5]` would be a better choice for example than `[3, 4, 5]`, because it shows that the coordinate values might be different, but they may overlap.

Reply:
> As I understand xarray notation, `x` is not a coordinate but a dimension.

I don't think their terminology is very good.  It seems to me that there are multiple things:

1. A dimension, which is a property of a data variable
2. A dimension definition, which is a mapping of dimension name to value -- a dimension definition is what I believe an xarray `Coordinate` object refers to (this is sometimes what they mean when they say "coordinate").
3. A coordinate value ("the x coordinate is 3").

I find their use of these terms imprecise and confusing.  It seems lie the same thing, `x` is sometimes referred to as a "dimension" and sometimes a "coordinate." This makes me 🤪

Reply:
I don't quite follow the part about the posterior. Most of the time, the variables in the posterior and in posterior predictive will be different. They can share one or several dimensions amd the respective coordinates but I think this depends on the model, it is not something arviz can control. The posterior/posterior predictive relation is sharing samples, they will share chain and draw dimensions as well as their respective coordinates.

Reply:
The point about the posterior is to distinguish the relationship between the posterior and posterior predictive samples (must have the same dimensions and coordinates) and between the posterior and (out of sample) predictions (may have different coordinates, but the same dimensions).

Reply:
I don't think the posterior and the posterior predictive must share dimensions and coordinates. Take the linear regression used as schema example for instance. The posterior has 3 variables (intercept, slope and sigma) and 2 dimensions (chain and draw), whereas the posterior predictive has one variable (slack_comments) and 3 dimensions (chain, draw, developer)

Reply:
Regarding terminology of dimension and coordinate it could be interesting to add a terminology section. I will reread xarray terminology section to see if we can base it on theirs. I think we only need a subset of xarray functionality (dimensions with multidimensional coordinates could be interesting, but I don't see the use of dimensions without coordinates nor of coordinates without dimensions) so let's hope we can achieve something we all agree and understand in the same way.

Reply:
Right, that's correct, because the posterior predictive by default contains the observed RVs only, and those are in `observed_data` rather than `posterior` group.

But you aren't allowed to change the existing shape of these observed RVs in the posterior predictive.

I was thinking of this in terms of the posterior, because the `observed_data` group is created as a side-effect of creating the `posterior` group.

Reply:
Oohh, I see now. I think this is in the posterior predictive section, though it may not be very well explained.

Maybe we could include "matching samples" and "matching variables" in the terminology